File: app/src/main/java/de/bikebean/app/ui/status/menu/history/HistoryAdapter.java
package de.bikebean.app.ui.status.menu.history;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import de.bikebean.app.db.DatabaseEntity;

public abstract class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder> {

    public abstract class HistoryViewHolder extends RecyclerView.ViewHolder {
        public HistoryViewHolder(View v) {
            super(v);
        }
    }

    protected Context ctx;
    protected final LayoutInflater mInflater;
    protected List<? extends DatabaseEntity> mStates;

    public HistoryAdapter(Context context, List<? extends DatabaseEntity> states) {
        ctx = context;

        mInflater = LayoutInflater.from(context);
        mStates = states;
    }

    void setStates(List<? extends DatabaseEntity> states) {
        mStates = states;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        if (mStates != null)
            return mStates.size();
        else return 0;
    }
}

File: app/src/main/java/de/bikebean/app/ui/drawer/status/history/HistoryAdapter.kt
package de.bikebean.app.ui.drawer.status.history

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.bikebean.app.db.DatabaseEntity
import de.bikebean.app.ui.drawer.status.history.HistoryAdapter.HistoryViewHolder

abstract class HistoryAdapter(
        protected val ctx: Context,
        states: List<DatabaseEntity?>?
) : RecyclerView.Adapter<HistoryViewHolder?>() {

    abstract class HistoryViewHolder(v: View) : RecyclerView.ViewHolder(v)

    protected val mInflater: LayoutInflater = LayoutInflater.from(ctx)
    protected var mStates: List<DatabaseEntity?>? = states

    fun setStates(states: List<DatabaseEntity?>?) {
        mStates = states
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int = mStates?.size ?: 0

}